File: CodingSamples/DotNet/Fundamentals/Runtime/ReflectionTest/BankFinLib/Finance/HomeLoan.cs
namespace Finance;

public class HomeLoan
{
    public float Common(double amount, int period) => amount < 5000000 ? 8.5f : 8.0f;

    public float Woman(double amount, int period) => Common(amount, period) - 1;

    public float Welfare(double amount, int period) => 0.6f * Common(amount, period);
}

File: CodingSamples/Java/Fundamentals/Runtime/ReflectionTest/bankfin/src/main/java/finance/HomeLoan.java
package finance;

public class HomeLoan {

    public float common(double amount, int period) {
        return amount < 5000000 ? 8.5f : 8;
    }

    public float women(double amount, int period) {
        return common(amount, period) - 1;
    }

    @MaxDuration(12) //value=12
    public float welfare(double amount, int period) {
        return 0.6f * common(amount, period);
    }


}
